pkg/querier/queryrange/results_cache.go

```diff
@@ -182,6 +182,7 @@ func shouldCacheResponse(r Response) bool {
 			for _, v := range hv.Values {
 				if v == noCacheValue {
 					shouldCache = false
+					level.Debug(s.logger).Log("msg", fmt.Sprintf("%s header in response is equal to %s, not caching the response", cachecontrolHeader, noCacheValue))
 					break outer
 				}
 			}
```

Comment:
```suggestion
				if v == noCacheValue {
					level.Debug(s.logger).Log("msg", fmt.Sprintf("%s header in response is equal to %s, not caching the response", cachecontrolHeader, noCacheValue))
					return false
				}
			}
```
(also get rid of `shouldCache`)